libpyvinyl/Parameters/Parameter.py

```diff
@@ -8,126 +8,168 @@ class Parameter(AbstractBaseClass):
+                print(
+                    "         This option is declared differently w.r.t. to the previous ones"
+                )
+                # should it throw an expection?
+                raise ValueError("Parameter", "options", "multiple validities")
+
         if isinstance(option, list):
             self.options += option
         else:
             self.options.append(option)
 
-    def set_value(self, value):
+    def set_option_as_legal(self, mybool=True):
+        self.options_are_legal = mybool
+
+    @property
+    def value(self):
+        return self._value
+
+    @value.setter
+    def value(self, value):
         """
         Sets value of this parameter if value is legal, otherwise warning is shown
 
         This could be expanded to raise an exception, or such could be in is_legal
         """
         if self.is_legal(value):
-            self.value = value
+            self._value = value
         else:
-            print("WARNING: Value of parameter '" + self.name
-                  + "' illegal, ignored.")
+            raise ValueError("Value of parameter '" + self.name + "' illegal.")
 
-    def is_legal(self, value=None):
+    def is_legal(self, values=None):
         """
         Checks whether or not given or contained value is legal given constraints.
 
         Illegal intervals have the highest priority to be checked. Then it will check the
         legal intervals and options. The overlaps among the constrains will be overridden by
         the constrain of higher priority.
         """
-        if value is None:
-            value = self.value
-
-        # Check illegal intervals
-        for illegal_interval in self.illegal_intervals:
-            if illegal_interval[0] < value < illegal_interval[1]:
-                return False
-
-        # Check legal intervals
-        is_inside_a_legal_interval = False
-        for legal_interval in self.legal_intervals:
-            if legal_interval[0] < value < legal_interval[1]:
-                is_inside_a_legal_interval = True
 
-        if not is_inside_a_legal_interval and len(self.legal_intervals) > 0:
-            return False
+        truths = []
+        if values is None:
+            values = self._value
 
-        # checked intervals, can return if options not used (frequent case)
-        if len(self.options) == 0:
-            return True
+        if not hasattr(values, "__iter__") or isinstance(values, str):
+            #            print(str(hasattr(values, "__iter__")) + str(values))
+            value = values
 
-        for option in self.options:
-            if option == value:
-                # If the value matches any option, it is legal
-                return True
+            for option in self.options:
+                if option == value:
+                    #
+                    return self._options_are_legal
+
+            # Check illegal intervals
```

Comment:
```suggestion
            # Check intervals
```

Reply:
Comment completely removed implementing the suggestion of splitting the checks with two methods, one for options and one for intervals.

Reply:
Comment completely removed implementing the suggestion of splitting the checks with two methods, one for options and one for intervals.